compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -1970,8 +1970,8 @@ private STNode parseParamList(boolean isParamNameOptional) {
                     paramsList.add(param);
                 } else {
                     paramOrderErrorPresent = true;
-                    updateLastNodeInListWithInvalidNode(paramsList, paramEnd, paramOrderError);
-                    updateLastNodeInListWithInvalidNode(paramsList, param, null);
+                    updateLastNodeInListWithInvalidNode(paramsList, paramEnd, null);
```

Comment:
Not related to this PR bt it would be better if we can overload this method without `DiagnosticCode`.

Reply:
I think the reason for not having an overloaded method is, if anyone wants to skip the diagnostic error code, they have to explicitly give the parameter as null.
 
If we create an overloaded method, one could easily forget to add the diagnostics when updating with invalid nodes.

Reply:
+1